### PR TITLE
Better error message for incompatible default argument (take 2)

### PR DESCRIPTION
**Pull request by elazarg**
Reopen #3773

### CODE REVIEW

**Comment by elazarg** [comment]
@gvanrossum

File: mypy/checker.py [hunk 1]
```diff
@@ -740,9 +740,15 @@ def is_implicit_any(t: Type) -> bool:
...
                 # Type check initialization expressions.
                 for arg in item.arguments:
-                    init = arg.initialization_statement
-                    if init:
-                        self.accept(init)
+                    if arg.initializer is not None:
+                        name = arg.variable.name()
+                        msg = 'Incompatible default for '
+                        if name.startswith('__tuple_arg_'):
+                            msg += "tuple argument no' {}".format(name[12:])
```
Comment (gvanrossum) [inline mypy/checker.py]: Few people will understand `argument no' 1` -- just get rid of the  `no'` and make it `argument 1` please.

**Comment by gvanrossum** [comment]
Thanks! Next time it would be nice to have an issue before you start coding. :-)

**Comment by elazarg** [comment]
Will do.

**Pull request merged**